### PR TITLE
Fix sending empty metric tags for failed requests

### PR DESCRIPTION
**Pull request by imiric**
This fixes an issue introduced in #1570 where metrics for failed requests were being sent with missing `url`, `method` and `name` tags, causing requests to show up with "0" tags in the cloud.

### CODE REVIEW

File: lib/netext/httpext/transport.go [hunk 1]
```diff
@@ -130,9 +130,6 @@ func (t *transport) measureAndEmitMetrics(unfReq *unfinishedRequest) *finishedRe
 			}
 		}
 
-		if enabledTags.Has(stats.TagMethod) {
-			tags["method"] = unfReq.request.Method
-		}
```
Comment (MStoykov) [inline lib/netext/httpext/transport.go:135]: I think this needs to stay ... as I am pretty sure if you have `POST -- 301 redirect --> GET` the second request will still be tagged as  `POST` ... I think

Comment (imiric) [reply lib/netext/httpext/transport.go]: OK, reverted. We should have a test for this, but I don't want to delay the canary further.

Comment (MStoykov) [reply lib/netext/httpext/transport.go]: Can you start on the test at least, we are not in that much hurry ;)

File: lib/netext/httpext/request_test.go [hunk 2]
```diff
@@ -199,6 +202,20 @@ func TestMakeRequestTimeout(t *testing.T) {
...
+	sampleCont := <-samples
+	allSamples := sampleCont.GetSamples()
+	require.Len(t, allSamples, 8)
+	expTags := map[string]string{
+		"error":      "context deadline exceeded",
+		"error_code": "1000",
+		"status":     "0",
+		"method":     "GET",
+		"url":        srv.URL,
+		"name":       "",
```
Comment (MStoykov) [inline lib/netext/httpext/request_test.go:214]: Shouldn't this be equal to `url` ?

Comment (imiric) [reply lib/netext/httpext/request_test.go:214]: Not according to the `preq.URL.Name != preq.URL.Clean()` check. If I set `Name: srv.URL` on line 196, then the `name` tag is skipped entirely. If I set it to something else, then it does get tagged with the custom value.

Since this is a unit test, I think this is the expected behavior when the `name` tag is not set manually.

File: lib/netext/httpext/transport.go (hunk 1 above)
Comment (imiric) [reply lib/netext/httpext/transport.go]: I don't think it's worth delaying the release for this one specific edge case test, but sure, I'll add it.

File: lib/netext/httpext/request_test.go (hunk 2 above)
Comment (MStoykov) [reply lib/netext/httpext/request_test.go:214]: Yeah ... but it should have `name` ... and it should be equal to the `url`.
Otherwise, it will ... just not have a `name` at all. I would actually expect that in this case `name` isn't set at all

Comment (imiric) [reply lib/netext/httpext/request_test.go:214]: Yes, `name` is not set if I set `Name: srv.URL`. Would you prefer that change?

File: lib/netext/httpext/transport.go (hunk 1 above)
Comment (imiric) [reply lib/netext/httpext/transport.go]: Resolved by af0f5f0c.

You were right, removing this would've broken the second request's `method` tag. 👍

File: lib/netext/httpext/request_test.go (hunk 2 above)
Comment (MStoykov) [reply lib/netext/httpext/request_test.go:214]: I just want every request metric to have `name`, `url` and `method`, just as before ;) 
Currently if the request is with status `0` for some reason and doesn't get to the transport it will not have `name` the same way that this actually fixes this for the other tags (`method` and `url`) it should fix it for `name`

Comment (imiric) [reply lib/netext/httpext/request_test.go:214]: > I just want every request metric to have name, url and method, just as before ;)

This PR fixes that. The current behavior of an empty `name` tag (note: not missing, but _empty_) is a different issue which already existed even before #1570. You can confirm this by backporting this test change to e.g. `v0.27.0`.

The issue you fixed in #1634 is separate to this fix, but I think it also makes sense. So let's merge #1634, and then this PR?

**Review by MStoykov — APPROVED** [review]
